sentry_sdk/integrations/celery.py

```diff
@@ -89,20 +89,30 @@ def _inner(*args, **kwargs):
         with hub.push_scope() as scope:
             scope._name = "celery"
             scope.clear_breadcrumbs()
-            _continue_trace(args[3].get("headers") or {}, scope)
+            span = _continue_trace(args[3].get("headers") or {}, scope)
+
+            with capture_internal_exceptions():
```

Comment:
This looks like magic. Why would assigning `task.name` to `scope.transaction` be `capture_internal_exceptions`?

Reply:
I honestly don't remember at this point, but I assume it's because celery's task objects are garbage. Added a comment